source/fs/fsutil.c

```diff
@@ -525,11 +525,9 @@ bool PathCopyVrtToFat(char* dest, char* orig, u32* flags) {
         }
     }
     
-    // check destination write permission (special paths only)
-    if (((*dest == '1') || (strncmp(dest, "0:/Nintendo 3DS", 16) == 0)) &&
-        (!flags || (*flags & (OVERWRITE_CUR|OVERWRITE_ALL))) &&
-        (!flags || !(*flags & OVERRIDE_PERM)) && 
-        !CheckWritePermissions(dest)) return false;
+    // check destination write permission
+    if ((!flags || !(*flags & OVERRIDE_PERM)) && !CheckWritePermissions(dest))
+        return false;
     
```

Comment:
A quick note, just in case you wonder. I took over the commit before this to my GM9 repo, and part of this commit.

The reason why I didn't take over this (f.e.) is because what you removed here is actually a somewhat hacky solution to speed things up, especially when copying large numbers of files. The addditional checks in here make sure that we only check write permissions when it actually makes sense (on drive 1:/ and in the 0:/Nintendo 3DS folder).

If you ever intend to do a pull request, unsure about the features that come in the commits after this. Can be discussed, though. If this was mainly for your personal use, don't bother ;).